File: fixtures/queue-consumer-closure/scripts/backfill.ts
// A one-off script run by hand. It is inside the functions' CodeUri,
// and no handler imports it.

export async function backfillRefunds(ids: string[]): Promise<void> {
  for (const id of ids) {
    await fetch("https://billing.example.internal/v1/refunds", {
      method: "POST",
      body: JSON.stringify({ id }),
    });
  }
}

File: fixtures/queue-consumer-closure-python/scripts/backfill.py
# A one-off script run by hand. It is inside the functions' CodeUri,
# and no handler imports it.

import requests


def backfill_refund(refund_id):
    return requests.post(
        "https://billing.example.internal/v1/refunds", json={"id": refund_id}
    )
